File: 1512-number-of-good-pairs/main.cpp
class Solution {
public:
    int numIdenticalPairs(vector<int>& nums) {
        
        std::unordered_map<int, int> count_values;
        int count_good_pairs = 0;

        for (const int& num : nums) {
            if (count_values.find(num) != count_values.end()) {
                count_good_pairs += count_values[num];
            }
            count_values[num]++;
        }
        
        return count_good_pairs;
    }
};

File: 1512-number-of-good-pairs/main.go
func numIdenticalPairs(nums []int) int {

    countValues := make(map[int]int)

    var countGoodPairs int = 0
    for _, num := range nums {
        countGoodPairs += countValues[num]
        countValues[num]++
    }
    
    return countGoodPairs
}
